Doc/library/imaplib.rst

```diff
@@ -307,6 +308,97 @@ An :class:`IMAP4` instance has the following methods:
    of the IMAP4 QUOTA extension defined in rfc2087.
 
 
+.. method:: IMAP4.idle(dur=None)
+
+   Return an :class:`!Idler`: an iterable context manager implementing the ``IDLE``
+   command as defined in :rfc:`2177`.
+
+   The context manager sends the ``IDLE`` command when activated by the
+   :keyword:`with` statement, produces IMAP untagged responses via the
+   :term:`iterator` protocol, and sends ``DONE`` upon context exit.
+
+   The *dur* argument sets a maximum duration (in seconds) to keep idling,
+   after which iteration will stop.  It defaults to ``None``, meaning no time
+   limit.  Callers wishing to avoid inactivity timeouts on servers that impose
+   them should keep this at most 29 minutes.
+   See the :ref:`warning below <windows-pipe-timeout-warning>` if using
+   :class:`IMAP4_stream` on Windows.
+
+   Response tuples produced by the iterator almost exactly match those
+   returned by other methods in this module.  The difference is that the tuple's
+   second member is a single response datum, rather than a list of data.
+   Therefore, in a mailbox where calling ``M.response('EXISTS')`` would
+   return ``('EXISTS', [b'1'])``, the idle iterator would produce
+   ``('EXISTS', b'1')``.
+
+   Example::
+
+      with M.idle(dur=29 * 60) as idler:
+          for typ, datum in idler:
+              print(typ, datum)
+
+      ('EXISTS', b'1')
+      ('RECENT', b'1')
+
+   Instead of iterating one response at a time, it is also possible to retrieve
+   the next response along with any immediately available subsequent responses
+   (e.g. a rapid series of ``EXPUNGE`` events from a bulk delete).  This
+   batch processing aid is provided by the context's ``burst()``
```

Comment:
I think we can still link to methods.

Reply:
Sorry for not responding quickly. I'd like some sort of reference, regardless of whether its hyperlinked or not; `burst()` without context of the class can be confusing for users.

Reply:
> Ah, yes, that does get sphinx to make a working link. However, it also burdens the reader with a bunch of unimportant namespacing, and cements where Idler is defined (which we do not want, for the same reasons that it is a deliberately undocumented class).

You can change how the link is being rendered as follows:

```rst
:meth:`Idler.burst <imaplib.IMAP4.Idler.burst>
```

This will render `Idler.burst` but will use the correct link.